pipeline/psth_foraging.py

```diff
@@ -0,0 +1,602 @@
+        """
+        Retrieve / build data needed for a Unit PSTH Plot based on the given
+        unit condition and included / excluded condition (sub-)variables.
+        Returns a dictionary of the form:
+          {
+             'trials': ephys.Unit.TrialSpikes.trials,
+             'spikes': ephys.Unit.TrialSpikes.spikes,
+             'psth': UnitPsth.unit_psth,
+             'raster': Spike * Trial raster [np.array, np.array]
+          }
+        """
+        # from sys import exit as sys_exit  # NOQA
+        # from code import interact
+        # from collections import ChainMap
+        # interact('unitpsth make', local=dict(ChainMap(locals(), globals())))
+
+        trials = TrialCondition.get_func(condition_key)()
+ 
+        unit_psth = (UnitPsth & {**condition_key, **unit_key}).fetch1('unit_psth')
+        if unit_psth is None:
+            raise Exception('No spikes found for this unit and trial-condition')
+
+        spikes, trials = (ephys.Unit.TrialSpikes & trials & unit_key).fetch(
+            'spike_times', 'trial', order_by='trial asc')
+
+        raster = [np.concatenate(spikes),
+                  np.concatenate([[t] * len(s)
+                                  for s, t in zip(spikes, trials)])]
+
+        return dict(trials=trials, spikes=spikes, psth=unit_psth, raster=raster)
+
+
+@schema
+class Selectivity(dj.Lookup):
+    """
+    Selectivity lookup values
+    """
+
+    definition = """
+    selectivity: varchar(24)
+    """
+
+    contents = zip(['contra-selective', 'ipsi-selective', 'non-selective'])
+
+
+@schema
+class PeriodSelectivity(dj.Computed):
+    """
+    Multi-trial selectivity for a specific trial subperiod
+    """
+
+    definition = """
+    -> ephys.Unit
+    -> experiment.Period
+    ---
+    -> Selectivity.proj(period_selectivity='selectivity')
+    ipsi_firing_rate=0:           float  # mean firing rate of all ipsi-trials
+    contra_firing_rate=0:         float  # mean firing rate of all contra-trials
+    p_value=1:                    float  # all trial spike rate t-test p-value
+    """
+
+    alpha = 0.05  # default alpha value
+
+    key_source = experiment.Period * (ephys.Unit & ephys.ProbeInsertion.InsertionLocation & 'unit_quality != "all"')
+
+    def make(self, key):
+        '''
+        Compute Period Selectivity for a given unit.
+        '''
+        log.debug('PeriodSelectivity.make(): key: {}'.format(key))
+
+        hemi = _get_units_hemisphere(key)
+
+        # retrieving the spikes of interest,
+        spikes_q = ((ephys.Unit.TrialSpikes & key)
+                    * (experiment.BehaviorTrial
+                       & {'task': 'audio delay',
+                          'early_lick': 'no early',
+                          'outcome': 'hit',
+                          'free_water': 0,
+                          'auto_water': 0})
+                    & (experiment.TrialEvent & 'trial_event_type = "delay"' & 'duration = 1.2')
+                    - experiment.PhotostimEvent)
```

Comment:
Looks like this code block is still about the `delay response` task. Is this `PeriodSelectivity` and the downstream `UnitSelectivity` table used yet?
If not we can remove these code blocks and add in later when needed